File: dbflow/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/UniqueModel.java
package com.raizlabs.android.dbflow.test.sql;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ConflictAction;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.annotation.Unique;
import com.raizlabs.android.dbflow.annotation.UniqueGroup;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description:
 */
@Table(database = TestDatabase.class,
       uniqueColumnGroups = {@UniqueGroup(groupNumber = 1, uniqueConflict = ConflictAction.ROLLBACK),
               @UniqueGroup(groupNumber = 2, uniqueConflict = ConflictAction.REPLACE)})
public class UniqueModel extends BaseModel {

    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    @Unique(uniqueGroups = 1, onUniqueConflict = ConflictAction.REPLACE)
    String uniqueName;

    @Column
    @Unique(uniqueGroups = 2, onUniqueConflict = ConflictAction.ROLLBACK)
    String anotherUnique;

    @Column
    @Unique(uniqueGroups = {1, 2})
    String sharedUnique;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/sql/unique/UniqueModel.kt
package com.raizlabs.android.dbflow.test.sql.unique

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ConflictAction
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.annotation.Unique
import com.raizlabs.android.dbflow.annotation.UniqueGroup
import com.raizlabs.android.dbflow.structure.BaseModel
import com.raizlabs.android.dbflow.test.TestDatabase

/**
 * Description:
 */
@Table(database = TestDatabase::class, uniqueColumnGroups = arrayOf(UniqueGroup(groupNumber = 1, uniqueConflict = ConflictAction.IGNORE), UniqueGroup(groupNumber = 2, uniqueConflict = ConflictAction.REPLACE)))
class UniqueModel : BaseModel() {

    @Column
    @PrimaryKey(autoincrement = true)
    var id: Long = 0

    @Column
    @Unique(uniqueGroups = intArrayOf(1), onUniqueConflict = ConflictAction.REPLACE)
    var uniqueName: String? = null

    @Column
    @Unique(uniqueGroups = intArrayOf(2), onUniqueConflict = ConflictAction.IGNORE)
    var anotherUnique: String? = null

    @Column
    @Unique(uniqueGroups = intArrayOf(1, 2))
    var sharedUnique: String? = null
}
